internal/pkg/deploy/cloudformation/stack/lb_web_svc.go

```diff
@@ -60,62 +60,75 @@ type LoadBalancedWebService struct {
 	parser loadBalancedWebSvcReadParser
 }
 
-// NewLoadBalancedWebServiceOpts contains fields to configure LoadBalancedWebService.
-type NewLoadBalancedWebServiceOpts struct {
-	App                    *config.Application
-	Env                    *config.Environment
-	Manifest               *manifest.LoadBalancedWebService
-	RuntimeConfig          RuntimeConfig
-	RootUserARN            string
-	PublicSubnetCIDRBlocks []string
+// LoadBalancedWebServiceOption is used to configuring an optional field for LoadBalancedWebService.
+type LoadBalancedWebServiceOption func(s *LoadBalancedWebService)
+
+// WithNLB enables Network Load Balancer in a LoadBalancedWebService.
+func WithNLB(cidrBlocks []string) func(s *LoadBalancedWebService) {
+	return func(s *LoadBalancedWebService) {
+		s.publicSubnetCIDRBlocks = cidrBlocks
+	}
+}
+
+// LoadBalancedWebServiceConfigs contains fields to configure LoadBalancedWebService.
+type LoadBalancedWebServiceConfigs struct {
+	App           *config.Application
+	Env           *config.Environment
+	Manifest      *manifest.LoadBalancedWebService
+	RuntimeConfig RuntimeConfig
+	RootUserARN   string
 }
 
 // NewLoadBalancedWebService creates a new CFN stack with an ECS service from a manifest file, given the options.
-func NewLoadBalancedWebService(opts NewLoadBalancedWebServiceOpts) (*LoadBalancedWebService, error) {
+func NewLoadBalancedWebService(conf LoadBalancedWebServiceConfigs,
+	opts ...LoadBalancedWebServiceOption) (*LoadBalancedWebService, error) {
 	parser := template.New()
-	addons, err := addon.New(aws.StringValue(opts.Manifest.Name))
+	addons, err := addon.New(aws.StringValue(conf.Manifest.Name))
 	if err != nil {
 		return nil, fmt.Errorf("new addons: %w", err)
 	}
 	var dnsDelegationEnabled, httpsEnabled bool
 	var appInfo deploy.AppInformation
 
-	if opts.App.DNSDelegated() {
+	if conf.App.Domain != "" {
 		dnsDelegationEnabled = true
 		appInfo = deploy.AppInformation{
-			Name:                opts.App.Name,
-			DNSName:             opts.App.Domain,
-			AccountPrincipalARN: opts.RootUserARN,
+			Name:                conf.App.Name,
+			Domain:              conf.App.Domain,
+			AccountPrincipalARN: conf.RootUserARN,
 		}
 		httpsEnabled = true
 	} else {
```

Comment:
Was it intentional to wrap it with an `else` block here, for semantic purpose?

Reply:
It is intentional because if it has domain then `httpsEnabled` must be true and we don't have to check if it has certificates imported.